frontend/web/components/modals/CreateMetadataField.tsx

```diff
@@ -0,0 +1,338 @@
+import React, { FC, useEffect, useState } from 'react'
+import Utils from 'common/utils/utils'
+import InputGroup from 'components/base/forms/InputGroup'
+import Button from 'components/base/forms/Button'
+import SupportedContentTypesSelect from 'components/metadata/SupportedContentTypesSelect'
+
+import {
+  useCreateMetadataFieldMutation,
+  useGetMetadataFieldQuery,
+  useUpdateMetadataFieldMutation,
+} from 'common/services/useMetadataField'
+
+import { useGetSupportedContentTypeQuery } from 'common/services/useSupportedContentType'
+
+import {
+  useCreateMetadataModelFieldMutation,
+  useUpdateMetadataModelFieldMutation,
+  useDeleteMetadataModelFieldMutation,
+} from 'common/services/useMetadataModelField'
+import { MetadataModelField } from 'common/types/responses'
+
+type CreateMetadataFieldType = {
+  id?: string
+  isEdit: boolean
+  metadataModelFieldList?: MetadataModelField[]
+  onComplete?: () => void
+  organisationId: string
+  projectId?: string
+}
+type IsRequiredForType = {
+  content_type: number
+  object_id: string
+}
+
+type QueryBody = {
+  content_type: string | number
+  field: number
+  is_required_for: boolean | IsRequiredForType[]
+}
+
+type Query = {
```

Comment:
Are these types all for requests? If so, does this mean they are duplicates of types that are in requests.ts

Reply:
Corrected.